packages/components/src/internal/components/domainproperties/ScannableOption.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, memo, useCallback } from 'react';
+import { Col, FormControl, Row } from 'react-bootstrap';
+import { DomainFieldLabel } from './DomainFieldLabel';
+import { createFormInputId, createFormInputName } from './actions';
+import { DOMAIN_FIELD_SCANNABLE_OPTION } from './constants';
+import { isFieldFullyLocked } from './propertiesUtil';
+import { ITypeDependentProps } from './models';
+
+export interface ScannableProps extends ITypeDependentProps{
+    scannable?: boolean;
+    appPropertiesOnly?: boolean;
+    showScannableOption?: boolean;
+}
+
+export const ScannableOption: FC<ScannableProps> = memo(props => {
+    const { domainIndex, index, lockType, scannable = false, appPropertiesOnly, showScannableOption, onChange } = props;
+    if (!appPropertiesOnly || !showScannableOption) return undefined;
```

Comment:
this likely doesn't matter, but I've see more `return null` in this case then I've seen `return undefined`

Reply:
updated